src/views/Farms/Farms.tsx

```diff
@@ -159,58 +157,18 @@ const Farms: React.FC = () => {
     }
   }
 
-  // /!\ This function will be removed soon
-  // This function compute the APY for each farm and will be replaced when we have a reliable API
-  // to retrieve assets prices against USD
   const farmsList = useCallback(
     (farmsToDisplay: Farm[]): FarmWithStakedValue[] => {
-      const cakePriceVsBNB = new BigNumber(farmsLP.find((farm) => farm.pid === CAKE_POOL_PID)?.tokenPriceVsQuote || 0)
       let farmsToDisplayWithAPY: FarmWithStakedValue[] = farmsToDisplay.map((farm) => {
-        if (!farm.tokenAmount || !farm.lpTotalInQuoteToken) {
+        if (!farm.lpTotalInQuoteToken || !prices) {
           return farm
         }
-        const cakeRewardPerBlock = CAKE_PER_BLOCK.times(farm.poolWeight)
-        const cakeRewardPerYear = cakeRewardPerBlock.times(BLOCKS_PER_YEAR)
-
-        // cakePriceInQuote * cakeRewardPerYear / lpTotalInQuoteToken
-        let apy = cakePriceVsBNB.times(cakeRewardPerYear).div(farm.lpTotalInQuoteToken)
-
-        if (farm.quoteToken.symbol === tokens.busd.symbol || farm.quoteToken.symbol === tokens.ust.symbol) {
-          apy = cakePriceVsBNB.times(cakeRewardPerYear).div(farm.lpTotalInQuoteToken).times(bnbPrice)
-        } else if (farm.quoteToken.symbol === tokens.eth.symbol) {
-          apy = cakePrice.div(ethPriceUsd).times(cakeRewardPerYear).div(farm.lpTotalInQuoteToken)
-        } else if (farm.quoteToken.symbol === tokens.cake.symbol) {
-          apy = cakeRewardPerYear.div(farm.lpTotalInQuoteToken)
-        } else if (farm.dual) {
-          const cakeApy =
-            farm && cakePriceVsBNB.times(cakeRewardPerBlock).times(BLOCKS_PER_YEAR).div(farm.lpTotalInQuoteToken)
-          const dualApy =
-            farm.tokenPriceVsQuote &&
-            new BigNumber(farm.tokenPriceVsQuote)
-              .times(farm.dual.rewardPerBlock)
-              .times(BLOCKS_PER_YEAR)
-              .div(farm.lpTotalInQuoteToken)
-
-          apy = cakeApy && dualApy && cakeApy.plus(dualApy)
-        }
-
-        let liquidity = farm.lpTotalInQuoteToken
 
-        if (!farm.lpTotalInQuoteToken) {
-          liquidity = null
-        }
-        if (farm.quoteToken.symbol === tokens.wbnb.symbol) {
-          liquidity = bnbPrice.times(farm.lpTotalInQuoteToken)
-        }
-        if (farm.quoteToken.symbol === tokens.cake.symbol) {
-          liquidity = cakePrice.times(farm.lpTotalInQuoteToken)
-        }
+        const quoteTokenPriceUsd = prices[farm.quoteToken.symbol.toLowerCase()]
```

Comment:
Should we add a check in case the API does not have the farm symbol? Or a fallback number.

Reply:
In case we don't have a number, `quoteTokenPriceUsd` will be null, then the apy will be `NaN` and this case is handled by the function used to get the apy, it will just return null. The FE will keep showing the loader.

Reply:
BTW, my next step in this rework is to remove the api, and compute the price from the farms data, so we won't have any missing data